packages/functional-tests/lib/metrics.ts

```diff
@@ -161,4 +161,15 @@ export class MetricsObserver {
     }
     return params;
   }
+
+  // Added as part of FXA-9322 to resolve flaky bug
+  // Can be removed once underlying issue to inconsistencies in recorded metrics flow events
+  // has been determined and resolved upstream
+  removeMultipleInitialViewEvents(eventTypes: (string | undefined)[]) {
+    if (eventTypes[0] === eventTypes[1]) {
+      eventTypes.shift();
+    }
```

Comment:
I think we can make this method a little safer:
1) Can we make sure the eventTypes[i] === 'amplitude.subPaySetup.view', In the unlikely case that the 0 and 1 index aren't equal to 'amplitude.subPaySetup.view' AND are equal
2) Can we handle 1 to many adjacent initial views? what if there are 3 for instance?

more specific and should probably handle 1-many adjacent

Reply:
Alternatively, since you are comparing the tail of the actual array, instead of removing duplicate initial views, perhaps you could slice the actual array. Something like....

```
      expect(
        actualEventTypes.slice(
          actualEventTypes.length - expectedEventTypes.length
        )
      ).toMatchObject(expectedEventTypes);
```

Reply:
Good call!

Added a check to ensure only event types that start with `amplitude.subPaySetup` are filtered in `actualEventTypes`.

Also took the alternative approach (as per our Slack conversation - complete with comment) and removed the verify method I added. 👍